framework/src/mfem/materials/MFEMGenericFunctorMaterial.C

```diff
@@ -0,0 +1,40 @@
+#ifdef MFEM_ENABLED
+
+#include "MFEMGenericFunctorMaterial.h"
+#include "MFEMProblem.h"
+
+registerMooseObject("MooseApp", MFEMGenericFunctorMaterial);
+
+InputParameters
+MFEMGenericFunctorMaterial::validParams()
+{
+  InputParameters params = MFEMMaterial::validParams();
+  params.addClassDescription("Declares material scalar properties based on names and coefficients "
```

Comment:
So these functor material coefficients essentially just wrap other coefficients?

Reply:
Yes. This is a way to use functions, variables, postprocessors, etc. as material properties.

Reply:
But since functions, variables, postprocessors are all themselves coefficients/functors is there any advantage to converting them to another type of coefficient/functor (materials)?

Reply:
Yes, because that allows you to build piece-wise functors across different blocks of the mesh.